File: practica1/src/gestorAplicacion/obras/Revista.java
package gestorAplicacion.obras;

import java.util.ArrayList;

public class Revista extends Publicacion {
	//ATRIBUTOS DE CLASE
	private static final long serialVersionUID = 1L;
	static public enum Meses { ENERO, FEBRERO,MARZO,ABRIL,MAYO,JUNIO,JULIO,AGOSTO,SEPTIEMBRE,OCTUBRE,NOVIEMBRE,DICIEMBRE};	
	static private ArrayList<Revista>revistas=new ArrayList<Revista>();
	
	//ATRIBUTOS INSTANCIA
	private short numero;
	private Meses mes;
	private String temporada;
	
	//CONSTRUCTORES
	public Revista (int codigo, String nombre, short año, short ejemplar, short numero,Meses mes, String temporada, Estanteria estanteria) {
		super(codigo,nombre,año,ejemplar);
		this.numero=numero;
		this.mes=mes;
		this.temporada=temporada;
		super.estanteria=estanteria;
		revistas.add(this);
		estanteria.getPublicaciones().add(this);
		Publicacion.numeroPublicaciones++;
	}
		
	//METODOS
	public String mostrarInfo() {
		return " INFORMACION DE LA REVISTA " + "\n" + "Nombre: " + this.nombre + "\n" + "Año: " + this.año + "\n" 
				+ "Mes: "+ this.mes + "\n"+ "Temporada: "+ this.temporada + "\n"
				+ "Codigo: " + this.codigo + "\n" + "Ejemplar: " + this.ejemplar + "\n" + "Estado: " + this.estado;
	}

	public String mostrarUbicacion() {
		return "Se localiza en -> " + this.estanteria.mostrarInfo();
	}
	
	
	
	//GETTERS Y SETTERS
	public short getNumero() {
		return numero;
	}

	public void setNumero(short numero) {
		this.numero = numero;
	}

	public Meses getMes() {
		return mes;
	}

	public void setMes(Meses mes) {
		this.mes = mes;
	}

	public String getTemporada() {
		return temporada;
	}

	public void setTemporada(String temporada) {
		this.temporada = temporada;
	}
	public static ArrayList<Revista> getRevistas() {
		return revistas;
	}
	public static void setRevistas(ArrayList<Revista> revistas) {
		Revista.revistas = revistas;
	}
	

}

File: practica2-python/src/gestorAplicacion/obras/Revista.py
from gestorAplicacion.obras.Publicacion import Publicacion,Estado
# from Publicacion import Estado
from enum import Enum

class Meses:
    jan='ENERO'
    feb='FEBRERO'
    mar='MARZO'
    apr='ABRIL'
    may="MAYO"
    jun="JUNIO"
    jul="JULIO"
    ago="AGOSTO"
    sep="SEPTIEMBRE"
    oct="OCTUBRE"
    nov="NOVIEMBRE"
    dec="DICIEMBRE"

class Revista(Publicacion): 
    
    # serialversion ?

    #Atributos de clase
    
    _revista = []
    

    #Constructor
    
    def __init__(self, codigo,nombre,año,ejemplar,numero=0,mes='...', temporada=0,estanteria=None,prestamo=None) :
        Revista._revista.append(self)
        super().__init__(codigo,nombre,año,ejemplar)
        self._numero = numero
        self._mes=mes
        self._temporada=temporada
        self.estanteria = estanteria
        self.prestamo = prestamo
        
        
        if estanteria is not None:
            self.estanteria.getPublicaciones.append(self)

    #Metodos
    
    def mostrarInfo(self):
        return  f'INFORMACION DE LA REVISTA: \n Nombre: {self.getNombre()} \n Año: {self.getAño()} \n Mes: {self.getMes()} \n Temporada: {self.getTemporada()} \n Codigo: {self.getCodigo()} \n Ejemplar: {self.getEjemplar()} \n Estado: {self.getEstado()} \n '
        
    def mostrarUbicacion(self):
        return "Localizada en: " + str(self.estanteria.mostrarInfo())
    
    def getNumero(self):
        return self._numero
    def setNumero(self,a):
        self._numero = a

    def getMes(self):
        return self._mes
    def setMes(self,a):
        self._mes = a
    
    def getTemporada(self):
        return self._temporada
    def setTemporada(self,a):
        self._temporada=a

    @classmethod
    def getRevista(cls):
        return Revista._revista
    @classmethod
    def setRevista(cls,lista):
        Revista._revista = lista